src/battle-text-parser.ts

```diff
@@ -529,6 +545,18 @@ class BattleTextParser {
 			return line1 + template.replace('[POKEMON]', this.pokemon(pokemon)).replace('[MOVE]', move);
 		}
 
+		case 'canDynamax': {
+			let [, side] = args;
+			const own = this.own(side);
+			let template = '';
+			if (this.turn === 1) {
+				if (own) template = this.template('canDynamax', own);
+			} else {
+				template = this.template('canDynamax', own);
+			}
```

Comment:
Can you explain what the turn tracker is about?

Reply:
Looks like it displays a message unless it's turn 1 and it's not yourself? What's the message and why?

Reply:
In gen 8 multi battles, there are two different dynamax messages:
```
canDynamax: "  [TRAINER] can dynamax now!",
canDynamaxOwn: "Dynamax Energy gathered around [TRAINER]!",
```
The "Energy" message is only used on the first turn a player is allowed to dynamax, and only for that player. The other players just see the "can" message, but on the first turn, only the players who can dynamax (the ones on their respective lefts) see any message at all.

Reply:
Server-side, this is done by sending the canDynamax messages as secret messages to their respective sides, but the server and client don't support a super-secret message protocol that only gets sent to one player on a team, so the client has to do a bit of work on its own.

Reply:
Just saying, request.active[0].canDynamax is true if that side can dynamax, otherwise its not defined.

Reply:
Okay, but protocol messages have never been mixed-case.

I also don't think `canDynamax` is a major message. I would go with `-candynamax`.